.jenkinsci/bindings.groovy

```diff
@@ -1,27 +1,69 @@
 #!/usr/bin/env groovy
 
-def doBindings() {
-  def cmake_options = ""
-  if (params.JavaBindings) {
-    cmake_options += " -DSWIG_JAVA=ON "
-  }
-  if (params.PythonBindings) {
-    cmake_options += " -DSWIG_PYTHON=ON "
-  }
-  // In case language specific options were not set,
-  // build for each language
-  if (!params.JavaBindings && !params.PythonBindings) {
-    cmake_options += " -DSWIG_JAVA=ON -DSWIG_PYTHON=ON "
-  }
+def doJavaBindings(buildType=Release) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/java-bindings-%2$s-%3$s-%4$s.zip', 
+    [currentPath, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
   sh """
     cmake \
       -H. \
       -Bbuild \
-      -DCMAKE_BUILD_TYPE=Release \
-      ${cmake_options}
+      -DCMAKE_BUILD_TYPE=$buildType \
+      -DSWIG_JAVA=ON
+  """
+  sh "cd build; make -j${params.PARALLELISM} irohajava"
+  sh "zip -j $artifactsPath build/shared_model/bindings/*.java build/shared_model/bindings/libirohajava.so"
+  sh "cp $artifactsPath /tmp/bindings-artifact"
+  return artifactsPath
+}
+
+def doPythonBindings(buildType=Release) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/python-bindings-%2$s-%3$s-%4$s.zip', 
+    [currentPath, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
+  sh """
+    cmake \
+      -H. \
+      -Bbuild \
+      -DCMAKE_BUILD_TYPE=$buildType \
+      -DSWIG_PYTHON=ON \
+      -DSUPPORT_PYTHON2=ON
   """
   sh "cmake --build build --target python_tests"
-  sh "cd build; make -j${params.PARALLELISM} irohajava irohapy"
+  sh "cd build; make -j${params.PARALLELISM} irohapy"
+  sh "zip -j $artifactsPath build/shared_model/bindings/*.py build/shared_model/bindings/*.so"
+  sh "cp $artifactsPath /tmp/bindings-artifact"
+  return artifactsPath
+}
+
+def doAndroidBindings(abiVersion) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/android-bindings-%2$s-%3$s-%4$s-%5$s-%6$s.zip', 
+    [currentPath, "\$PLATFORM", abiVersion, "\$BUILD_TYPE_A", sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
+  sh """
+    (cd /iroha; git init; git remote add origin https://github.com/hyperledger/iroha.git; \
+    git fetch --depth 1 origin develop; git checkout -t origin/develop)
+  """
+  sh """
+    . /entrypoint.sh; \
+    sed -i.bak "s~find_package(JNI REQUIRED)~#find_package(JNI REQUIRED)~" /iroha/shared_model/bindings/CMakeLists.txt; \
```

Comment:
That better be done as separate file, like a patch or so. Otherwise changes in `android-build.sh` require changes here

Reply:
Agree. I think it should look like a single patch that is applicable both for `android-build.sh` and pipeline script. @luckychess what do you think?

Reply:
Sounds good. But I do not like idea of the patch because every minor change of original CMakeLists.txt will break everything. May be a separate bash script - but can it will be called from here?

Reply:
It could be a bash script. But the thing is that variables used in that script must be properly escaped. Otherwise, Jenkins interprets them as Groovy variables (see `entrypoint.sh` in this commit). With that said, it would require to maintain two separate versions of the script: one for self-building and one for CI. The idea with the patch is more tangible solution as for me.